extensions/metadata/metadata.go

```diff
@@ -0,0 +1,64 @@
+package metadata
+
+import (
+	"bufio"
+	"strings"
+
+	"github.com/emad-elsaid/xlog"
+	. "github.com/emad-elsaid/xlog"
+	"gopkg.in/yaml.v2"
+)
+
+func init() {
+	RegisterPreprocessor(stripYAML)
+}
+
+func stripYAML(content Markdown) Markdown {
+	reader := strings.NewReader(string(content))
+
+	scanner := bufio.NewScanner(reader)
+	var body strings.Builder
+	var y strings.Builder
+	inFrontMatter := false
+	frontMatterFound := false
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if strings.TrimSpace(line) == "---" {
```

Comment:
I think the frontmatter is anyline that has more than 3 `-` not exactly 3

Reply:
I actually don't really know. I also didn't found any "standard", but took this as reference: https://jekyllrb.com/docs/front-matter/
Actually, one could do this "flexible" and allow the user to have more than 3. That is now an implementation flavor :)